src/api/spec/controllers/webui/project_controller_spec.rb

```diff
@@ -687,4 +687,34 @@
       is_expected.to redirect_to(project_show_path(user.home_project))
     end
   end
+
+  describe 'POST #unlock' do
+    before do
+      login user
+    end
+
+    it "project can't be unlocked if it hasn't lock and enable flags" do
+      post :unlock, project: user.home_project
+      is_expected.to redirect_to(action: :show, project: user.home_project)
+      expect(flash[:notice]).to eq("Project can't be unlocked: is not locked")
+    end
+
+    it "maintenance incident can't be unlocked if there are maintenance release requests" do
+      user.home_project.flags.create(flag: 'lock', status: 'enable')
```

Comment:
And another context for the opposite with ``it "project can't be unlocked if it hasn't lock and enable flags" do`` in it! That also reduces the length of the test name and would improve the readability...